src/lib/services/assetService.ts

```
@@ -1,19 +1,58 @@
+  } = useCelatoneApp();
+  const registryAssets = registry?.assets ?? [];
+
   return useQuery<AssetInfos>(
     [CELATONE_QUERY_KEYS.ASSET_INFOS, assetsApiRoute, withPrices],
     async () =>
-      getAssetInfos(assetsApiRoute, withPrices).then((assets) =>
-        assets.reduce((acc, asset) => ({ ...acc, [asset.id]: asset }), {})
-      ),
+      getAssetInfos(assetsApiRoute, withPrices).then((assets) => {
+        const assetsMap = assets.reduce<AssetInfos>(
+          (acc, asset) => ({ ...acc, [asset.id]: asset }),
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)